src/hooks/use-with-google-doc-versions.ts

```diff
@@ -73,14 +75,16 @@ export function useWithStoreDocVersions(selectedActivityId: string) {
         docData.plainText === lastUpdatedPlainText &&
         docData.title === lastUpdatedTitle &&
         messages.length === lastNumMessages &&
-        sessionId === lastSavedSessionId
+        sessionId === lastSavedSessionId &&
+        selectedActivityId === lastSavedActivityId
```

Comment:
there are too many checks here, so probably wrap them in a `Boolean` to always get a correct value and could move them to a separate variable

Reply:
applied, good call